### ISSUE TITLE
Why IoU calculation is (I + I)/(I +U)

### ISSUE THREAD

**Issue by evakasch** [megvii-research/PETR#44]
The function of IoU in mmdet3d_plugin/models/detectors/petr3d_seg.py seems to be (I + I)/(I +U), but not we common used I/U.
This may cause the IoU result higher than normal IoU metrics. So why not use normal IoU metrics? 

The source code is shown below:

def IOU (intputs,targets):
    numerator = 2 * (intputs * targets).sum(dim=1)
    denominator = intputs.sum(dim=1) + targets.sum(dim=1)
    loss = (numerator + 0.01) / (denominator + 0.01)
    return loss

**Comment by vaesl** [comment]
Thanks for your pointing out this issue. Indeed, there is a silly bug in calculating the IoU metric. We will first update the Readme to remind the users of PETRv2 BEV segmentation. After that, we will fix this issue and update the IoU performances in both the PETR repo and PETRv2 paper on the arxiv. Thanks again for your kind reminder!

**Comment by aoyanl** [comment]
> Thanks for your pointing out this issue. Indeed, there is a silly bug in calculating the IoU metric. We will first update the Readme to remind the users of PETRv2 BEV segmentation. After that, we will fix this issue and update the IoU performances in both the PETR repo and PETRv2 paper on the arxiv. Thanks again for your kind reminder!

Hi，Have you fixed this issue in code and paper? I am doubted about the iou announced in paper v3. The iou announced in Table 3 is higher than that in v2 and v1, but according to the discussion above, the value should be lower？

**Comment by Zhao-Qihao** [comment]
@vaesl hi, have you updated the  IoU metric in PETRv2?
